From ce930557895a86a527bed4fee7ca23caeed0ec61 Mon Sep 17 00:00:00 2001
From: Yipeng Zou <zouyipeng@huawei.com>
Date: Thu, 4 May 2023 16:53:52 +0800
Subject: [PATCH 1276/1646] [Huawei] genirq: introduce handle_fasteoi_edge_irq
 for phytium

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6WPFT

--------------------------------

Since we fix the issue that irq may lose in gic-v3.

It also needs fixed in phytium platform.

Fixes: 5715a76fcc5b ("[Huawei] genirq: introduce handle_fasteoi_edge_irq flow handler")
Signed-off-by: Yipeng Zou <zouyipeng@huawei.com>
---
 drivers/irqchip/irq-gic-phytium-2500.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-phytium-2500.c b/drivers/irqchip/irq-gic-phytium-2500.c
index a0c622fb2039..ffaf267ad14f 100644
--- a/drivers/irqchip/irq-gic-phytium-2500.c
+++ b/drivers/irqchip/irq-gic-phytium-2500.c
@@ -1526,7 +1526,7 @@ static int gic_irq_domain_map(struct irq_domain *d, unsigned int irq,
 		if (!gic_dist_supports_lpis())
 			return -EPERM;
 		irq_domain_set_info(d, irq, hw, chip, d->host_data,
-				    handle_fasteoi_irq, NULL, NULL);
+				    handle_fasteoi_edge_irq, NULL, NULL);
 		break;
 
 	default:
-- 
2.17.1

